### ISSUE TITLE
Current driver does not support Vertex Buffer Object.

### ISSUE THREAD

**Issue by timjim333** [WhitewaterFoundry/Fedora-Remix-for-WSL#186]
**Describe the bug**
Starting the [OpenVSP](https://github.com/OpenVSP/OpenVSP/blob/main/README.md) graphical app `vsp` leads to a crash that may be mesa-related - the output is copy-pasted below. Apologies if this is not a WSL bug - I#m unsure if this is related to the being [unable to update to the latest mesa](https://github.com/WhitewaterFoundry/Fedora-Remix-for-WSL/issues/180)

**To Reproduce**
1. Compile VSP
2. Start XLaunch in Windows
3. run ./vsp

**Expected behavior**
For the graphical app to launch.

**Screenshots**
```
$ /opt/openvsp/OpenVSP-3.31.1-Linux/vsp
Current driver does not support Vertex Buffer Object.
OpenGL version reports: 1.4 (3.3 (Compatibility Profile) Mesa 21.0.0-devel (git-8f342ce8b0))
Current driver does not support Vertex Buffer Object.
OpenGL version reports: 1.4 (3.3 (Compatibility Profile) Mesa 21.0.0-devel (git-8f342ce8b0))
Current driver does not support Vertex Buffer Object.
OpenGL version reports: 1.4 (3.3 (Compatibility Profile) Mesa 21.0.0-devel (git-8f342ce8b0))
Segmentation fault
```

**Additional context**

**Basic Troubleshooting Checklist**

[ x ] I have searched Google for the error message.
[ x ] I have checked official WSL troubleshooting documentation: https://docs.microsoft.com/en-us/windows/wsl/troubleshooting#confirm-wsl-is-enabled.
[ ] I have searched the official Microsoft WSL issues page: https://github.com/Microsoft/WSL/issues.
[ ] I have searched the WLinux issues page: https://github.com/WhitewaterFoundry/WLinux/issues.
[ ] I have reset WLinux: Settings->Apps->Apps & features->WLinux->Advanced Options->Reset. 
[ ] I have disabled and re-enabled WSL in Windows Features.
[ ] I have run Windows 10 updates and restarted.

What other troubleshooting have you attempted?

Running dnf update, removing and reinstalling mesa-libGL-devel.

**Fedora Remix for WSL Version**

NAME="Fedora Remix for WSL"
VERSION="37"
ID=fedoraremixforwsl
ID_LIKE=fedora
VERSION_ID=37
PLATFORM_ID="platform:f37"
PRETTY_NAME="Fedora Remix for WSL"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:37"
HOME_URL="https://github.com/WhitewaterFoundry/Fedora-Remix-for-WSL"
SUPPORT_URL="https://github.com/WhitewaterFoundry/Fedora-Remix-for-WSL"
BUG_REPORT_URL="https://github.com/WhitewaterFoundry/Fedora-Remix-for-WSL/issues"
PRIVACY_POLICY_URL="https://github.com/WhitewaterFoundry/Fedora-Remix-for-WSL/blob/master/PRIVACY.md"
FEDORA_REMIX_VERSION=37.0.4

**Windows Build**

OS Name:                   Microsoft Windows 10 Home
OS Version:                10.0.19045 N/A Build 19045
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
BIOS Version:              American Megatrends Inc. G751JT.213, 11/06/2019

**Comment by crramirez** [comment]
Hello @timjim333

Be sure that the variable LIBGL_ALWAYS_INDIRECT is unset

`unset LIBGL_ALWAYS_INDIRECT`

Also run `glxinfo -B` and paste the results here

**Comment by timjim333** [comment]
@crramirez thanks for the reply! Unsetting `LIBGL_ALWAYS_INDIRECT` (or setting `LIBGL_ALWAYS_INDIRECT=0`) before running the progarm solved the issue for vsp - but it seemed to cause glxinfo to crash:

```
$ LIBGL_ALWAYS_INDIRECT=0 glxinfo -B
name of display: 172.22.96.1:0
display: 172.22.96.1:0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Microsoft Corporation (0xffffffff)
    Device: D3D12 (NVIDIA GeForce GTX 970M) (0xffffffff)
    Version: 21.2.3
    Accelerated: yes
    Video memory: 19382MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 3.3
    Max compat profile version: 3.1
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.0
OpenGL vendor string: Microsoft Corporation
OpenGL renderer string: D3D12 (NVIDIA GeForce GTX 970M)
OpenGL core profile version string: 3.3 (Core Profile) Mesa 21.2.3
OpenGL core profile shading language version string: 3.30
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 3.1 Mesa 21.2.3
OpenGL shading language version string: 1.40
OpenGL context flags: (none)

OpenGL ES profile version string: OpenGL ES 3.0 Mesa 21.2.3
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.00

Segmentation fault
```
The same thing happens after unset.
Whereas leaving it set:
```
$ LIBGL_ALWAYS_INDIRECT=0 glxinfo -B
name of display: 172.22.96.1:0
display: 172.22.96.1:0  screen: 0
direct rendering: No (LIBGL_ALWAYS_INDIRECT set)
OpenGL vendor string: Mesa/X.org
OpenGL renderer string: softpipe
OpenGL version string: 1.4 (3.3 (Compatibility Profile) Mesa 21.0.0-devel (git-8f342ce8b0))
```

Was this expected? Thanks!

**Comment by crramirez** [comment]
Yes this is expected. Now you can use your program with the help of your GPU.

But Fedora Remix should unset this variable automatically. Could you run this script `upgrade.sh` twice? To see if the em get solved 

Best Regards
